beard/clustering/wrappers.py

```diff
@@ -80,8 +86,17 @@ def __init__(self, method="single", affinity="euclidean", threshold=None,
             See scipy.cluster.hierarchy.fcluster for further details.
 
         :param scoring: callable
-            The scoring function to maximize in (semi-)supervised clustering
-            (when y!=None).
+            The scoring function to maximize in either (semi-)supervised
+            clustering if y is not None, or unsupervised otherwise (using
+            ``silhouette_score`` as an example when y is None or all y =-1).
+            In case of silhouette_score, 'affinity' parameter of this
+            function must be set to *precomputed*.
+            See sklearn.metrics.silhouette_score for further details.
```

Comment:
We should be explicit about the expected input arguments. 

We should state that the scoring function needs to have the following prototype:
`scoring(labels_true, labels_pred)`, `scoring(labels_pred)`, `scoring(Xs, labels_true, labels_pred)` or  `scoring(Xs, labels_pred)` depending on the case and the value of `affinity_score`.